### PR TITLE
Fix #25 OpenTripPlanner-based trip planning feature

### LINKED ISSUE #25: Trip planning
There's no longer an OBA API for it, but people continue to ask for it. Frankly I doubt I will ever have time to do this, but I want to keep something open because it's such a popular feature request.

To be clear, this would involve either bringing back Brian's API, integrating OpenTripPlanner, or running my own server. If it were a different server it would have to be kept updated with GTFS updates just like the main OBA server is.

Moved from: http://code.google.com/p/seattle-bus-bot/issues/detail?id=83

### PR DESCRIPTION
**Pull request by sdjacobs**
- Code was integrated from York OBA app and OpenTripPlanner-for-Android.
- New TripPlanActivity with TripPlanFragment and TripResultsFragment.
- If a region has an OpenTripPlanner instance specified in otpBaseUrl field,
that instance will be queried from the trip planner.
- Realtime feature: after a trip is planned and an itinerary is selected, OTP
will be requeried every minute. If the rank of the selected itinerary changes
in a returned list of itineraries, or the selected itinerary is no longer
present, the user will be notified and can replan the trip.

Supersedes #541

### CODE REVIEW

File: onebusaway-android/src/main/res/values-w820dp/dimens.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<resources>
+    <!-- Example customization of dimensions originally defined in res/values/dimens.xml
+         (such as screen margins) for screens with more than 820dp of available width. This
+         would include 7" and 10" devices in landscape (~960dp and ~1280dp respectively). -->
+    <dimen name="activity_horizontal_margin">64dp</dimen>
```
Comment (barbeau) [inline onebusaway-android/src/main/res/values-w820dp/dimens.xml]: Just curious - why was this added for w820dp devices?

**Comment by barbeau** [comment]
@sdjacobs Nice work on the Amazon build variant - worked first time launching on a Fire Phone, which is pretty impressive! :)

**Comment by barbeau** [comment]
@sdjacobs Thanks, this looks good!  Very excited to finally get trip planning into OBA :+1: :+1: :+1: .

@sdjacobs @cagryInside I'm going to merge this so I can get it out to beta, and we can open individual issues for resolving things that pop up.

File: onebusaway-android/src/main/res/values-w820dp/dimens.xml (hunk 1 above)
Comment (sdjacobs) [inline onebusaway-android/src/main/res/values-w820dp/dimens.xml]: Looks like you added this in the initial commit for 25-tripPlan (7293e443).
Maybe it was added automagically by Android Studio for some reason.

Comment (barbeau) [inline onebusaway-android/src/main/res/values-w820dp/dimens.xml]: Ah, right, must have been auto generated by AS.
